src/utils/type-transform/Payload.ts

```diff
@@ -46,26 +52,55 @@ function deconstructAlpha41(str: string, throws = false): null | never {
 }
 
 const versions = new Set([deconstructAlpha40_Or_Alpha39, deconstructAlpha41])
-export function deconstructPayload(str: string): Payload | null
-export function deconstructPayload(str: string, throws: true): Payload
-export function deconstructPayload(str: string, throws = false): Payload | null {
-    for (const ver of versions) {
-        const result = ver(str, false)
-        if (throws === false) return result
-        if (result) return result
-        return ver(str, true)
+
+/**
+ * @type Decoder    defines decoder. if null, auto select decoder.
+ */
+type Decoder = SocialNetworkWorkerAndUI['payloadDecoder'] | null
+type Encoder = SocialNetworkWorkerAndUI['payloadEncoder']
+
+export function deconstructPayload(str: string, decoder: Decoder, opts?: { throws: false }): Payload | null
+export function deconstructPayload(str: string, decoder: Decoder, opts?: { throws: true }): Payload
+export function deconstructPayload(
+    str: string,
+    decoder: Decoder,
+    opts: { throws?: boolean } = { throws: false },
```

Comment:
decoder is required, not optional.

Reply:
decoder is required, not optional.

Reply:
Then you can make opts required

Reply:
I still thought that decoder should be placed in argument list since it is an important input for deconstruct. Is there any style guide for something like this?

Reply:
no style guide for this. but it's some type of inconsisitency api design. `opts: {throws, decoder}` or `decoder, throws` will be fine but not the mix of them

Reply:
Note: will be resolved in stage 4